### ISSUE TITLE
add dataset module

### ISSUE THREAD

**Issue by solegalli** [feature-engine/feature_engine#535]
first function should be to load and prepare the titanic dataset from openML

**Comment by solegalli** [comment]
Sure @datacubeR 

In many of our user guide demos for categorical encoders, we use the titanic dataset from open_ml. 

We use variations of this function to import and prepare the data:

```
# Load dataset
def load_titanic():
        data = pd.read_csv('https://www.openml.org/data/get_csv/16826755/phpMYEkMl')
        data = data.replace('?', np.nan)
        data['cabin'] = data['cabin'].astype(str).str[0]
        data['pclass'] = data['pclass'].astype('O')
        data['embarked'].fillna('C', inplace=True)
        return data

data = load_titanic()

# Separate into train and test sets
X_train, X_test, y_train, y_test = train_test_split(
                data.drop(['survived', 'name', 'ticket'], axis=1),
                data['survived'], test_size=0.3, random_state=0)
```

If would be great if all of that, including removing the variables name and ticket and probably home as well, went into a function in a new module called datasets, so that in the demos, we can just do `load_titanic`, and it returns the predictors in X and the target in y as a dataframe and a series.

See for example this demo: https://feature-engine.readthedocs.io/en/latest/user_guide/encoding/CountFrequencyEncoder.html

The second step would be to update all the examples in the documentation to use this function instead of doing all the steps manually as we do now.

**Comment by solegalli** [comment]
closed here: #544
